Fixed Pause Menu & Movement

computer-kyle opened the pull request:
## Description

Finally fixed movement bug, and the pause menu is no longer contingent on the player having a RigidBody
## Type of Change

- [x] Bug Fix
- [ ] New Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please specify):

## Related Issue

Link to the issue this pull request addresses (if applicable):
- Closes #11 
- Closes #36 

## Changes Made

- Removed player RigidBody
- Added public static boolean to player Movement class, called LockMovement
- Changed references to RigidBody to utilize LockMovement in the pause menu script

## Checklist

- [x] Code follows the project's coding style.
- [ ] If asset(s) added, they have been made into a prefab
- [ ] Tests have been added/updated.
- [ ] Documentation has been updated (if applicable).
- [ ] No unused code or assets remain.
- [x] Unity project builds without errors.

## Screenshots (if applicable)

N / A

## Additional Notes

N / A

computer-kyle committed:
Fixed Pause Menu & Movement

- Finally fixed movement bug, and the pause menu is no longer contingent on the player having a RigidBody

uid106 reviewed (approved):
## Pull Request Review
- **Functionality**:
  - Fixed movement jittering.
  - Fixed pause menu contingency on player rigid body.
  - Adds movement class, LockMovement.

- **Unused Code**: 
  - No unused code.

- **Code Style**: 
   - Properly following project code style.

- **Script Structure**: 
   -  Script structure is well put together. 

- **Documentation**: 
   - Documentation is appropriate. 

- **Build Integrity**: 
  - Code compiles and runs with no issues.

---

### Summary
The script is functioning as intended, walking feels significantly better and the pause menu is functioning. Currently there is a samplescene conflict that'll need to be resolved before merging. Also I noticed that the image for the pause menu is not present on this fix, it may have accidently been removed or made invisible. 

![image](https://github.com/user-attachments/assets/8d18fa04-2abb-4173-a31f-651de6d493b6)
![image](https://github.com/user-attachments/assets/3247eb45-5223-499f-a801-851b4222573c)

brandon722 reviewed (approved):
## Pull Request Review
- **Functionality**:
  - Changed pause menu so it is no longer contingent on freezing the player's rigid body.
      - Adds movement class, LockMovement
  - Fixed the bug with movement where it would appear jittery.

- **Unused Code**: 
  - None

- **Code Style**: 
   - Properly following project code style.

- **Script Structure**: 
   -  Script structure is well put together. 

- **Documentation**: 
   - Documentation is good and easy to understand.

- **Build Integrity**: 
  - Code compiles and runs with no issues.

---

### Summary
The script is working as intended. Walking is smoother and the pause menu still works with the rigid body changes. The merge conflict has been fixed; this is ready to be merged.

computer-kyle committed:
Fixed main menu bug

Fixed bug with main menu movement not unlocking when entering from the pause menu

computer-kyle committed:
Merge branch 'main' into week4_fixes_items
